rust/lang/security/audit/dangerous-system-call.yaml

```diff
@@ -0,0 +1,18 @@
+rules:
+  - id: argument-injection
+    patterns:
+      - pattern-either:
+          - pattern: |
+              fn $FUNC(...) {
+               requests::$METHOD(...);
+               ...
+               Command::new(...);
+               ...
+                       }
```

Comment:
This feels like it should become a taint mode rule with proper data flow analysis. Is the intent to monitor user input (through a request method) flowing into a command?

This should give you a start:

```
mode: taint
pattern-sources: 
  - pattern: |
       requests::$METHOD(...)
pattern-sinks:
  - pattern-either: 
     - pattern: Command::new(...). ... .args(...)
     - pattern: Command::new(...)
```

Currently this rule is assuming:
- every method after `requests::` is bad. In reality am assuming some will be harmless and not actually return dangerous user input, you will want to put constraints on the $METHOD metavariable
- it is dangerous for this input to end up in `Command::new(...)` and this can be called with or without additional `args()` calls. 

It's likely to not be completed yet.

Additionally, I wonder what the difference is between this rule and `dangerous-exec-command`. It feels like these two rules are looking for the same patterns.